app/Domain/Integrations/Controllers/IntegrationController.php

```diff
@@ -140,41 +133,61 @@ public function destroyUrl(Request $request, string $id, string $urlId): Redirec
         );
     }
 
-    public function update(UpdateIntegrationRequest $request, string $id): RedirectResponse
+    public function updateIntegration(UpdateIntegrationRequest $request, string $id): void
     {
         $currentIntegration = $this->integrationRepository->getById(Uuid::fromString($id));
 
-        $updatedIntegration = UpdateIntegrationMapper::map($request, $currentIntegration);
-
-        $this->integrationRepository->update($updatedIntegration);
+        if($currentIntegration->name !== $request->get('integrationName') || $currentIntegration->description !== $request->get('description')) {
+            $updatedIntegration = UpdateIntegrationMapper::map($request, $currentIntegration);
 
-        return Redirect::route(
-            TranslatedRoute::getTranslatedRouteName($request, 'integrations.show'),
-            [
-                'id' => $id,
-            ]
-        );
+            $this->integrationRepository->update($updatedIntegration);
+        }
     }
 
-    public function updateUrls(UpdateIntegrationUrlsRequest $request, string $id): RedirectResponse
+    public function update(UpdateIntegrationUrlsRequest $request, string $id): RedirectResponse
     {
+        $integrationRequest = ['integrationName' => $request->input('integrationName'), 'description' => $request->input('description')];
+
+        $integrationUpdateRequest = new UpdateIntegrationRequest($integrationRequest);
+
+        $this->updateIntegration($integrationUpdateRequest, $id);
+
         $ids = array_map(
             fn ($url) => Uuid::fromString($url['id']),
             array_filter(
                 [
-                    $request->input('loginUrl'),
+                    ...($request->input('loginUrls') ?? []),
```

Comment:
I think in Auth0 there is only a single login URL

Reply:
Yes indeed loginUrl should be singular